alembic/operations/toimpl.py

```diff
@@ -95,13 +96,27 @@ def create_index(
     operations: "Operations", operation: "ops.CreateIndexOp"
 ) -> None:
     idx = operation.to_index(operations.migration_context)
-    operations.impl.create_index(idx)
+    kw = {}
+    if operation.if_not_exists is not None:
+        if not sqla_2:
+            raise NotImplementedError("SQLAlchemy 2.0+ required")
```

Comment:
I'm not sure what's the usual behavior here when alembinc is trying to do something not supported by the installed sqlalchemy version.

@zzzeek do you remember if there is a rule for this?

Reply:
I haven't found the rule but there is a similar case https://github.com/sqlalchemy/alembic/blob/f5e7fd6d72e0a4c61bb72e691d0e0acfb6c594de/alembic/operations/base.py#L520-L521